package NYT.NChunkClient.NProto;

import "yt/core/misc/guid.proto";
import "yt/ytlib/node_tracker_client/node.proto";
import "yt/ytlib/chunk_client/chunk_meta.proto";
import "yt/ytlib/chunk_client/chunk_info.proto";
import "yt/ytlib/object_client/master_ypath.proto";

////////////////////////////////////////////////////////////////////////////////

message TReqConfirm
{
    repeated fixed32 replicas = 1;
    required NChunkClient.NProto.TChunkMeta chunk_meta = 2;
    required NChunkClient.NProto.TChunkInfo chunk_info = 3;
}

message TRspConfirm
{ }

////////////////////////////////////////////////////////////////////////////////

message TReqSeal
{
    required TMiscExt info = 1;
}

message TRspSeal
{ }

////////////////////////////////////////////////////////////////////////////////

// CreateObjects extension.

message TReqCreateChunkExt
{
    extend NObjectClient.NProto.TReqCreateObjects
    {
        optional TReqCreateChunkExt create_chunk_ext = 100;
    }

    // Desired number of copies (including those created by background replication).
    optional int32 replication_factor = 1 [default = 1];

    // Minimum number of replicas to read during seal. Journal chunks only.
    optional int32 read_quorum = 2 [default = 0];
 
    // Minimum number of replicas to flush during write. Journal chunks only.
    optional int32 write_quorum = 3 [default = 0];

    // See NErasure::ECodec.
    optional int32 erasure_codec = 4 [default = 0];

    // Can this chunk be balanced to other nodes?
    optional bool movable = 5 [default = true];

    // Should this chunk be marked as vital?
    optional bool vital = 6 [default = false];

    // If given, the chunk is immediately attached to this chunk list.
    optional NYT.NProto.TGuid chunk_list_id = 7;
}

////////////////////////////////////////////////////////////////////////////////
